### PR TITLE
adjust converter APIs

### PR DESCRIPTION
**Pull request by zhu733756**
Signed-off-by: zhu733756 <talonzhu@yunify.com>

This pr aims to adjust converter APIs for better integrating with KubeSphere/QKCP backend engine.
Features：
- Add function ConvertToDashboard.
- Split the serialization process for converting the dashboard model to JSON data.
- Split the logic for converting JSON data to Yaml data.

/cc @benjaminhuo @junotx

### CODE REVIEW

File: tools/converter/dashboard_converter.go [hunk 1]
```diff
@@ -64,41 +64,58 @@ func (converter *Converter) ConvertFromJson(content []byte, isClusterCrd bool, n
...
 		Kind:       kind,
 		Metadata:   metadata,
 		Spec:       dashboard,
-	}
+	}, nil
 
+}
+
+// ConvertKubsphereDashboardFromJson converts the input json content to json bytes content
+func (converter *Converter) ConvertDashboardToJson(content []byte, isClusterCrd bool, ns string, name string) error {
```
Comment (junotx) [inline tools/converter/dashboard_converter.go]: The method name in the comment does not match the actual method name

File: tools/converter/dashboard_converter.go [hunk 2]
```diff
@@ -64,41 +64,58 @@ func (converter *Converter) ConvertFromJson(content []byte, isClusterCrd bool, n
...
 		return err
 	}
 
 	converter.OutputJson = convertedJson
+	return nil
+}
 
-	convertedYaml, err := yamlConverter.JSONToYAML(convertedJson)
+// ConvertKubsphereDashboardFromJson converts the input json content to yaml bytes content
+func (converter *Converter) ConvertDashboardToYaml(content []byte, isClusterCrd bool, ns string, name string) error {
```
Comment (junotx) [inline tools/converter/dashboard_converter.go]: The method name in the comment does not match the actual method name

File: tools/converter/dashboard_converter.go [hunk 3]
```diff
@@ -64,41 +64,58 @@ func (converter *Converter) ConvertFromJson(content []byte, isClusterCrd bool, n
...
 	}
 
 	converter.OutputYaml = convertedYaml
-
 	return nil
 }
 
-// ConvertKubsphereDashboardFromFile: this method converts the input json file to a bytes content
+// ConvertKubsphereDashboardFromFile converts the input json file to yaml/json bytes content
 func (converter *Converter) ConvertFromFile(input io.Reader, isClusterCrd bool, ns string, name string) error {
```
Comment (junotx) [inline tools/converter/dashboard_converter.go]: The method name in the comment does not match the actual method name

File: tools/converter/dashboard_converter.go [hunk 4]
```diff
@@ -107,7 +124,7 @@ func (converter *Converter) ConvertFromFile(input io.Reader, isClusterCrd bool,
 	return nil
 }
 
-// ConvertKubsphereDashboardManifests: this method converts to a k8s mainfest file
+// ConvertKubsphereDashboardManifests converts to a k8s mainfest file
 func (converter *Converter) ConvertToKubsphereDashboardManifests(input io.Reader, output io.Writer, isClusterCrd bool, ns string, name string) error {
```
Comment (junotx) [inline tools/converter/dashboard_converter.go]: The method name in the comment does not match the actual method name

**Review by junotx — CHANGES_REQUESTED** [review]

**Review by junotx — APPROVED** [review]
/lgtm

**Review by benjaminhuo — APPROVED** [review]

**Pull request merged**